lib/plugins/expressionPlugin.js

```diff
@@ -63,10 +65,16 @@ function mergeAttributes(attrList1, attrList2) {
         }
 
         if (!found) {
-            additionalItems.push(attrList2[i]);
+            if (isNGSILD) {
+                // eslint eqeqeq - deliberate double equals to include undefined.
+                if (!(Number.isNaN(attrList2[i].value) || attrList2[i].value == null)) {
+                    additionalItems.push(attrList2[i]);
+                }
+            } else {
+                additionalItems.push(attrList2[i]);
+            }
```

Comment:
This is the core of this change of the PR and **fundamental** to change the processing. Given that `typeInformation` is already available to the JEXL plugin can you explain why `typeInformation` should not be used here?

Reply:
The problem is NGSI-LD specific code is included within a plugin. For maintainability reasons, instead of creating dependencies across the codebase, is better to put specific protocol behaviour in their own files (in this case, since the null management is a NGSI-LD specific requirement, the suggestion is to move the code between L49-L50 and L68-L75 to `lib/services/ngsi/entities-NGSI-LD.js`)

Reply:
Fixed cbecc05

Reply:
Thanks! Now this PR is very atomic and clean, great!